ferretdb/ferretdb_test.go

```diff
@@ -83,6 +84,54 @@ func Example() {
 	// Output: mongodb://127.0.0.1:17027/
 }
 
+func TestFerretDBWithCustomLogger(t *testing.T) {
+	logger := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{
+		Level: slog.LevelInfo,
+	}))
+	f, err := ferretdb.New(&ferretdb.Config{
+		PostgreSQLURL: testutil.PostgreSQLURL(t),
+		ListenAddr:    "127.0.0.1:0",
+		StateDir:      t.TempDir(),
+		Logger:        logger,
+	})
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(testutil.Ctx(t))
+	done := make(chan struct{})
+
+	go func() {
+		f.Run(ctx)
+		close(done)
+	}()
+
+	uri := f.MongoDBURI()
+	require.NotEmpty(t, uri)
+
+	client, err := mongo.Connect(options.Client().ApplyURI(uri))
+	require.NoError(t, err)
+
+	err = client.Ping(ctx, nil)
+	require.NoError(t, err)
+
+	res := client.Database("admin").RunCommand(ctx, bson.D{{Key: "getLog", Value: "global"}})
+
+	var actual bson.D
+	err = res.Decode(&actual)
+	require.NoError(t, err)
+
+	require.Len(t, actual, 3)
+	require.Equal(t, "log", actual[0].Key)
```

Comment:
What does actual[0] look like? Current assertions do not test `getLog` retrieves logs from handler :thinking:. Let's make sure to test that logs from handler are fetched.

Reply:
The `actual["log"]` value is the log in FerretDB. Maybe we can use regexp to make sure that the value is the log fetched by handler like `.*Listening on TCP.*` (the first log output by FerretDB)? If you have a better way, plz tell me :)

Reply:
I think we can execute a command which would produce a log, for example like below command produces `WARN` level log with `no such command: 'nonExistentCommand'`.

```
RunCommand(ctx, bson.D{{Key: "nonExistentCommand", Value: 1}})
```

Then we can check that specific log in the test assertion. I think startup logs such as `.*Listening on TCP.*` works slightly differently and we want to make sure the logs produced by commands work as expected :hugs:.

Doing test this way would also test `WARN` log level can be set :hugs: